contracts/periphery/src/pool/weighted.rs

```diff
@@ -42,21 +233,43 @@ impl OsmosisPool for WeightedPool {
 
     fn swap_exact_amount_in(
         &mut self,
-        input_amount: cosmwasm_std::Coin,
+        input_amount: Coin,
         output_denom: String,
-        min_output_amount: cosmwasm_std::Uint128,
+        min_output_amount: Uint128,
         spread_factor: Decimal,
     ) -> Result<Uint128, ContractError> {
-        todo!()
+        let amount_out = self.calc_out_amount_given_in(input_amount.clone(), output_denom.clone());
+
+        let new_pool_assets = self.generate_new_pool_assets(
+            input_amount.denom,
+            output_denom.clone(),
+            Uint128::zero() - input_amount.amount,
```

Comment:
ask; both variable ( Uint128::zero(), input_amount.amount) are `Uint128` isnt it? If right, `generate_new_pool_assets` argument type must be change. Or change input value. This code must raise panic because of out of uint128 range. @steve806